### PR TITLE
[feat] replace tag 'a' by 'link' / transform app folder into pages in order to import GetServerSidesProps

### PR DESCRIPTION
**Pull request by lucas-debize**

### CODE REVIEW

File: frontend/pages/_app.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+const MyApp: React.FC<AppProps> = ({ Component, pageProps }) => {
+  return (
+    <RootLayout>
+      <Component {...pageProps} />
+    </RootLayout>
+  );
+};
+
+export default MyApp;
```
Comment (alanschnegg) [inline frontend/pages/_app.tsx]: MyApp -> App

File: frontend/pages/component/RootLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const RootLayout: React.FC<RootLayoutProps> = ({ children }) => {
+  return (
+    <div>
+      <Head>
+        <title>Site Hub</title>
+        <link rel="icon" type="image/vnd.icon" href="/image/epitech_logo.ico" />
+      </Head>
+      <NavBar />
+      <div className="lg:ml-72 mt-16">
+        <div>{children}</div>
```
Comment (alanschnegg) [inline frontend/pages/component/RootLayout.tsx]: What's the use of this div ?

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: frontend/pages/component/RootLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import NavBar from './NavBar';
+import Head from 'next/head';
+
+const RootLayout: React.FC<RootLayoutProps> = ({ children }) => {
+  return (
+    <div>
+      <Head>
+        <title>Site Hub</title>
+        <link rel="icon" type="image/vnd.icon" href="/image/epitech_logo.ico" />
```
Comment (thmarinho) [inline frontend/pages/component/RootLayout.tsx]: rename the image into `favicon.ico`

File: frontend/pages/component/RootLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import NavBar from './NavBar';
+import Head from 'next/head';
+
+const RootLayout: React.FC<RootLayoutProps> = ({ children }) => {
+  return (
+    <div>
```
Comment (thmarinho) [inline frontend/pages/component/RootLayout.tsx]: Use a Fragment instead of a div

File: frontend/pages/project.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      <h1>Project</h1>
+      <p>{data}</p>
+    </div>
+  );
+};
+
+export default Project;
+
+export const getServerSideProps: GetServerSideProps = async () => {
+  const data = 'Données côté serveur';
```
Comment (thmarinho) [inline frontend/pages/project.tsx]: Don't send `Données côté serveur` but `[]`.
Later, you'll receive an array not a string

File: frontend/pages/project.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { GetServerSideProps } from 'next';
+
+const Project: React.FC<{ data: string }> = ({ data }) => {
```
Comment (thmarinho) [inline frontend/pages/project.tsx]: rename `data` into `projects`

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Review by thmarinho — APPROVED** [review]

File: frontend/pages/project.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { GetServerSideProps } from 'next';
+
+const Project: React.FC<{ projects: string[] }> = ({ projects }) => {
+  return (
+    <>
+      <h1>Project</h1>
+      <p>{projects}</p>
```
Comment (alanschnegg) [inline frontend/pages/project.tsx]: Projects is an array, html can't display arrays, you need to handle it manually for each object in the array

File: frontend/pages/component/RootLayout.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+const RootLayout: React.FC<RootLayoutProps> = ({ children }) => {
+  return (
+    <>
+      <Head>
+        <title>Site Hub</title>
+        <link rel="icon" type="image/vnd.icon" href="/image/favicon.ico" />
+      </Head>
+      <NavBar />
+      <div className="lg:ml-72 mt-16">
+        <div>{children}</div>
```
Comment (alanschnegg) [inline frontend/pages/component/RootLayout.tsx]: This div is useless

**Review by alanschnegg — CHANGES_REQUESTED** [review]

**Review by thmarinho — APPROVED** [review]

**Review by alanschnegg — APPROVED** [review]

**Pull request merged**